### DISCUSSION TITLE
Hiding implicit focus indicators

### DISCUSSION THREAD

**Discussion by meowgorithm** [charmbracelet/huh#592] [Ideas]
Having seen Huh in action for a bit, I think a good improvement would be to eliminate the focus indicators when they're implicit, with the goal of making the experience more visually focused and concise.

In other words, I don't think we always need to draw the little bar that indicates the active section in all cases.

Off the top of my head, the logic would work like this:

1. If the entire form is one single field, hide the indicator.
2. If the entire form has multiple groups with one single field in each group, hide all indicators.
3. In all other cases show the indicator.

**Comment by dez11de** [comment]
Creators of such forms that don't want the little bar can simply set theme.Focused.Base.BorderStyle to HiddenBorder.

**Comment by meowgorithm** [comment]
From a design perspective I'd prefer to default to hiding implicit indicators to reduce visual noise.

There's also a catch: focus indicators may not necessarily be borders: a theme could instead employ something different, such as a background color on the group, so manually adjusting borders won't necessarily do the trick.

If we find the former behavior is desired we could also add an option to enable it.

**Comment by bashbunni** [comment]
Hey @meowgorithm I've converted this to a discussion so we can talk through it and keep our backlog clear. If it does end up being something we want to add, we'll promote it back to an issue when we are ready to work on it.

I'm not sure when the focus indicator would be implicit, maybe it would be good to give an example case for future reference.
